tools/wpt/revlist.py

```diff
@@ -0,0 +1,299 @@
+import argparse
+import calendar
+import logging
+import os
+import subprocess
+import sys
+import time
+
+here = os.path.dirname(__file__)
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+logger = logging.getLogger()
+
+DEFAULT_EPOCH_STEP = 86400
+DEFAULT_EPOCH_THESHOLD = 600
+DEFAULT_EPOCH_UNTIL = calendar.timegm(time.gmtime())
+DEFAULT_HEAD_REVISION = "origin/master"
+DEFAULT_LOG_LEVEL = 30
+DEFAULT_NUM_REVISIONS = 1
+DEFAULT_VERBOSE = False
+
+REV_LIST_COUNT = 10000
+
+TAGGED_REVISIONS = "merge_pr_"
+
+WEEKLY_EPOCH_SIZE = 604800
+WEEKLY_EPOCH_OFFSET = -259200  # Monday, 29 December 1969 0:00:00
+
+
+MYPY = False
+if MYPY:
+    # MYPY is set to True when run under Mypy.
+    from typing import Any
+    from typing import Callable
+    from typing import Dict
+    from typing import List
+    from typing import Text
+    from typing import Union
+
+
+def parser_debug_log_level(**kwargs):
+    # type: (**Any) -> Any
+    debug_level = kwargs.get("log_level", DEFAULT_LOG_LEVEL)
+
+    if isinstance(debug_level, int):
+        return debug_level
+
+    try:
+        return int(debug_level)
+    except Exception:
+        pass
+
+    return debug_level
+
+
+def parser_epoch_step(**kwargs):
+    # type: (**Any) -> int
+    epoch_step = kwargs.get("epoch_step", DEFAULT_EPOCH_STEP)
+    if isinstance(epoch_step, int):
+        return epoch_step
+    if epoch_step.lower() == "three-hourly":
+        return 10800
+    if epoch_step.lower() == "six-hourly":
+        return 21600
+    if epoch_step.lower() == "twelve-hourly":
+        return 43200
+    if epoch_step.lower() == "daily":
+        return 86400
+    if epoch_step.lower() == "weekly":
+        return 604800
+    return int(epoch_step)
+
+
+def parser_epoch_until(**kwargs):
+    # type: (**Any) -> int
+    epoch_until = kwargs.get("epoch_until", DEFAULT_EPOCH_UNTIL)
+    if isinstance(epoch_until, int):
+        return epoch_until
+    if epoch_until.lower() == "now":
+        return DEFAULT_EPOCH_UNTIL
+    return int(epoch_until)
+
+
+def parser_num_revisions(**kwargs):
+    # type: (**Any) -> int
+    num_revisions = kwargs.get("num_revisions", DEFAULT_NUM_REVISIONS)
+    return int(num_revisions)
+
+
+COMMAND_ARGS = {
+    "epoch_until": {
```

Comment:
I don't think exposing this as an argument is necessary. The main reason to use it would be to list only revisions that come before HEAD, but even that I don't think we need for the initial version of the command. I've commented elsewhere on how we could do it later.